### PR TITLE
[MKL-DNN]conv2d transpose

### PR DESCRIPTION
**Pull request by jczaja**
This PR is introducing implementation of forward pass of conv2d transpose using MKL-DNN deconvolution primitive. 

**Notes:**
- It works only for is_test=true and should not be used in training
- Relevant Unit test works using NCHW (Any layout not tested yet)

@luotao1 If you have a model of interest that is using conv2d transpose then please test if this model works with this PR & share your observations.

### CODE REVIEW

File: paddle/fluid/operators/conv_transpose_mkldnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+      int h = iohw_weights_tz[2];
+      int w = iohw_weights_tz[3];
+      std::shared_ptr<T> reordered_filter_data(new T[o * c * h * w](),
+                                               std::default_delete<T[]>());
+      for (int i = 0; i < c; ++i) {
+        for (int j = 0; j < o; ++j) {
+          int in_offset = j * h * w + i * o * h * w;
+          int out_offset = j * c * h * w + i * h * w;
+          std::memcpy(&(reordered_filter_data.get())[out_offset],
+                      &filter_data[in_offset], h * w * sizeof(T));
```
Comment (tpatejko) [inline paddle/fluid/operators/conv_transpose_mkldnn_op.cc:101]: Lines 100 and 101 are a bit unreadable. You could rewrite it into:
```
auto reordered_filter = reordered_filter_data.get();
// for-loops
{ 
    int in_offset = ...;
    int out_offset = ...;

    std::copy_n(filter_data+in_offset, h*w, reordered_filter+out_offset);
}
```
What do you think?

**Comment by luotao1** [comment]
Thanks for your PR!
However, I test the model with this PR, there is no increase.
- command
```
export MKLDNN_VERBOSE=2
```
- result
```
Note: Google Test filter = Analyzer_resnet50.profile_mkldnn
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Analyzer_resnet50
[ RUN      ] Analyzer_resnet50.profile_mkldnn
Fused graph 0
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_Ohwi8o,num:1,64x3x7x7,0.136963
mkldnn_verbose,create,convolution,jit:avx2,forward_training,fsrc:nchw fwei:Ohwi8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic3oc64_ih207oh101kh7sh2dh0ph0_iw175ow85kw7sw2dw0pw0,0.097168
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_Ohwi8o,num:1,64x3x7x7,0.0212402
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nchw fwei:Ohwi8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic3oc64_ih207oh101kh7sh2dh0ph0_iw175ow85kw7sw2dw0pw0,4.17407
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,0.0661621
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,2.30981
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x64x101x85,0.112793
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,128x64x5x5,0.046875
mkldnn_verbose,create,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic64oc128_ih101oh49kh5sh2dh0ph0_iw85ow41kw5sw2dw0pw0,0.0998535
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x64x101x85,1.896
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,128x64x5x5,0.788086
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic64oc128_ih101oh49kh5sh2dh0ph0_iw85ow41kw5sw2dw0pw0,14.426
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.0488281
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.994873
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x128x49x41,0.0419922
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,256x128x5x5,0.0429688
mkldnn_verbose,create,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic128oc256_ih49oh23kh5sh2dh0ph0_iw41ow19kw5sw2dw0pw0,0.0949707
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x128x49x41,0.878906
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,256x128x5x5,3.09912
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic128oc256_ih49oh23kh5sh2dh0ph0_iw41ow19kw5sw2dw0pw0,13.709
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.0480957
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.361084
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x256x23x19,0.0771484
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,512x256x3x3,0.079834
mkldnn_verbose,create,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic256oc512_ih23oh11kh3sh2dh0ph0_iw19ow9kw3sw2dw0pw0,0.0969238
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x256x23x19,0.425049
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,512x256x3x3,4.55396
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic256oc512_ih23oh11kh3sh2dh0ph0_iw19ow9kw3sw2dw0pw0,4.59692
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.0930176
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.174072
mkldnn_verbose,create,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic512ih23iw19,0.0100098
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic512ih23iw19,1.30591
mkldnn_verbose,create,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic512ih23iw19,0.0600586
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic512ih23iw19,0.0710449
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x512x23x19,0.0380859
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x512x23x19,0.649902
mkldnn_verbose,create,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic256ih49iw41,0.00610352
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic256ih49iw41,2.9939
mkldnn_verbose,create,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic256ih49iw41,0.116211
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic256ih49iw41,0.151855
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x256x49x41,0.0571289
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x256x49x41,0.27002
mkldnn_verbose,create,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic128ih101iw85,0.027832
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic128ih101iw85,6.54785
mkldnn_verbose,create,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic128ih101iw85,0.059082
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic128ih101iw85,0.343994
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x128x101x85,0.0681152
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x128x101x85,0.929932
mkldnn_verbose,create,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_tanh,mb1ic3ih207iw175,0.078125
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_tanh,mb1ic3ih207iw175,0.13501
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_Ohwi8o,num:1,64x3x7x7,0.019043
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nchw fwei:Ohwi8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic3oc64_ih207oh101kh7sh2dh0ph0_iw175ow85kw7sw2dw0pw0,2.70605
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,0.0668945
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,0.712158
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x64x101x85,0.531006
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,128x64x5x5,0.251953
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic64oc128_ih101oh49kh5sh2dh0ph0_iw85ow41kw5sw2dw0pw0,13.196
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.052002
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.298096
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x128x49x41,0.23999
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,256x128x5x5,1.02197
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic128oc256_ih49oh23kh5sh2dh0ph0_iw41ow19kw5sw2dw0pw0,12.8691
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.0510254
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.101074
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x256x23x19,0.0959473
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_OIhw8i8o,num:1,512x256x3x3,1.43408
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic256oc512_ih23oh11kh3sh2dh0ph0_iw19ow9kw3sw2dw0pw0,4.41211
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.0869141
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.0449219
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic512ih23iw19,1.35278
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic512ih23iw19,0.0668945
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x512x23x19,0.0510254
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x512x23x19,0.175049
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic256ih49iw41,3.07715
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic256ih49iw41,0.165039
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x256x49x41,0.0800781
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x256x49x41,0.269043
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic128ih101iw85,6.56787
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic128ih101iw85,0.313965
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x128x101x85,0.0791016
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nchw,num:1,1x128x101x85,0.949951
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_tanh,mb1ic3ih207iw175,0.156006
[       OK ] Analyzer_resnet50.profile_mkldnn (551 ms)
[----------] 1 test from Analyzer_resnet50 (551 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (551 ms total)
[  PASSED  ] 1 test.
```
- command
```
./test_analyzer_resnet50 --gtest_filter=Analyzer_resnet50.profile_mkldnn --paddle_num_threads=4 --batch_size=1 --repeat=100 --infer_model=model/ --profile
```
- result
```
Place: CPU
Time unit: ms
Sorted by total time in descending order in the same thread
Event                        Calls       Total       Min.        Max.        Ave.        Ratio.
thread0::conv2d_transpose    400         6007.6      4.18907     41.8653     15.019      0.729454
thread0::conv2d              400         1192.36     0.858559    13.693      2.98091     0.144779
thread0::batch_norm          300         420.426     0.478607    11.4115     1.40142     0.051049
thread0::elementwise_add     400         278.552     0.161049    5.06228     0.696381    0.0338224
thread0::leaky_relu          400         245.635     0.237167    8.78435     0.614088    0.0298255
thread0::relu                300         56.7644     0.051408    4.9759      0.189215    0.00689245
thread0::tanh                100         21.5533     0.084525    2.97657     0.215533    0.00261704
thread0::fetch               100         12.1982     0.116327    0.167375    0.121982    0.00148113
thread0::feed                100         0.650486    0.0055      0.01508     0.00650486  7.89833e-05
```

**Comment by luotao1** [comment]
Maybe you should update
https://github.com/PaddlePaddle/Paddle/blob/6224e61fd94e6ad87f18c2808a76256b516fa3f3/paddle/fluid/framework/ir/is_test_pass.cc#L27-L37

File: paddle/fluid/operators/conv_transpose_mkldnn_op.cc [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+class ConvTransposeMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+ public:
+  void Compute(const paddle::framework::ExecutionContext& ctx) const override {
+    PADDLE_ENFORCE(paddle::platform::is_cpu_place(ctx.GetPlace()),
+                   "It must use CPUPlace.");
+
+    const bool is_test = ctx.Attr<bool>("is_test");
+    PADDLE_ENFORCE(
+        is_test == true,
+        "ConvTransposeMKLDNN works only for inference!. Set is_test = True");
```
Comment (luotao1) [inline paddle/fluid/operators/conv_transpose_mkldnn_op.cc:36]: You should update
https://github.com/PaddlePaddle/Paddle/blob/6224e61fd94e6ad87f18c2808a76256b516fa3f3/paddle/fluid/framework/ir/is_test_pass.cc#L27-L37
to make `is_test_pass` work on `ConvTransposeMKLDNN `

Comment (ghost) [reply paddle/fluid/operators/conv_transpose_mkldnn_op.cc:36]: I don't think we need update op_list.  The "conv_transpose" OP already have attribute "[is_test](https://github.com/PaddlePaddle/Paddle/blob/06d8777ec2857020da34b60ba359cf9b8e963b4e/paddle/fluid/operators/conv_transpose_op.cc#L113)". According to below code snippet, "is_set" should be already set to true.
https://github.com/PaddlePaddle/Paddle/blob/6224e61fd94e6ad87f18c2808a76256b516fa3f3/paddle/fluid/framework/ir/is_test_pass.cc#L41-L48

Comment (luotao1) [reply paddle/fluid/operators/conv_transpose_mkldnn_op.cc:36]: @jianhang-liu You are right, but why this PR does not increase the speed?

Comment (ghost) [reply paddle/fluid/operators/conv_transpose_mkldnn_op.cc:36]: I wonder ConvTransposeMKLDNNOpKernel had NOT been selected (and run). In MKLDNN log, no "deconv" primitive was displayed.

Comment (jczaja) [reply paddle/fluid/operators/conv_transpose_mkldnn_op.cc:36]: @luotao1 Here is output from MKLDNN verbose, for UT:
...
_mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:nchw fbia:undef fdst:nchw,alg:deconvolution_
...

I do not see deconvolution  entries in log you posted, so perhaps MKL-DNN op is **not** called.Please check (printf) why Conv2d_transpose MKL-DNN is not called

**Comment by luotao1** [comment]
I add a LOG in `mkldnn_placement_pass.cc`
```
diff --git a/paddle/fluid/framework/ir/mkldnn_placement_pass.cc b/paddle/fluid/framework/ir/mkldnn_placement_pass.cc
index 145a3a455..d7a338538 100644
--- a/paddle/fluid/framework/ir/mkldnn_placement_pass.cc
+++ b/paddle/fluid/framework/ir/mkldnn_placement_pass.cc
@@ -22,6 +22,8 @@ std::unique_ptr<ir::Graph> MKLDNNPlacementPass::ApplyImpl(
     std::unique_ptr<ir::Graph> graph) const {
   VLOG(30) << "Aplies MKL-DNN placement strategy.";
   for (const Node* n : graph->Nodes()) {
+    if (n->IsOp())
+      LOG(INFO) << n->Op()->Type() << ":" << n->Op()->HasAttr("use_mkldnn");
     if (n->IsOp() && n->Op()->HasAttr("use_mkldnn")) {
       n->Op()->SetAttr("use_mkldnn", true);
     }
```
And the result is:
```
I1128 03:42:48.564507 12909 mkldnn_placement_pass.cc:26] fetch:0
I1128 03:42:48.564527 12909 mkldnn_placement_pass.cc:26] elementwise_add:1
I1128 03:42:48.564568 12909 mkldnn_placement_pass.cc:26] conv2d_transpose:0
I1128 03:42:48.564575 12909 mkldnn_placement_pass.cc:26] relu:1
I1128 03:42:48.564591 12909 mkldnn_placement_pass.cc:26] tanh:1
I1128 03:42:48.564608 12909 mkldnn_placement_pass.cc:26] elementwise_add:1
I1128 03:42:48.564620 12909 mkldnn_placement_pass.cc:26] leaky_relu:0
```
The reason is that `conv2d_transpose` and `leaky_relu` don't have `use_mkldnn` attribute in model's program desc.

**Comment by jczaja** [comment]
@luotao1,

1. Could you please use this branch to recreate (train for 1 iteration) a model of your interest that is using conv2d transpose? That way we can have use_mkldnn attrib added to conv2d transpose , and performance of model can be evaluated or existance of other problems can checked.

2. Is it possible for us to fix that problem with missing use_mkldnn in your model, or you are planning to retrain that model? What is your suggesstions?

**Comment by luotao1** [comment]
@jczaja We fix the problem in #14659 , now `conv_transpose_mkldnn_op` works sucessfully. 
```
Place: CPU
Time unit: ms
Sorted by total time in descending order in the same thread

Event                        Calls       Total       Min.        Max.        Ave.        Ratio.
thread0::conv2d_transpose    400         4464.67     2.88586     29.9117     11.1617     0.687153
thread0::conv2d              400         1041.6      0.805595    12.9125     2.604       0.160311
thread0::batch_norm          300         376.649     0.480788    10.3276     1.2555      0.0579697
thread0::elementwise_add     400         288.708     0.163594    7.86578     0.72177     0.0444348
thread0::leaky_relu          400         247.529     0.236566    5.27947     0.618824    0.038097
thread0::relu                300         49.1335     0.054144    4.80587     0.163778    0.00756208
thread0::tanh                100         16.7112     0.074629    4.42513     0.167112    0.00257201
thread0::fetch               100         11.6969     0.112496    0.145033    0.116969    0.00180026
thread0::feed                100         0.64786     0.005602    0.018564    0.0064786   9.97114e-05
```
- `conv2d_transpose`: speedup 25%
- `conv2d`, `batch_norm` are all faster than before. The reason is that `is_test_pass` is not worked on `conv2d` and `batch_norm` since `is_test` attributes are not in the program desc.

**Comment by ghost** [comment]
@luotao1 Could you provide MKLDNN verbose log also? It will be useful for us to check which MKLDNN (internal) kernel was picked up when running conv_transpose OP. Thanks.

**Comment by luotao1** [comment]
```
I1128 21:34:58.212733 26727 helper.h:229] ====== batch_size: 1, repeat: 1, threads: 1, thread id: 0, latency: 359.939ms, fps: 2.77825 ======
I1128 21:34:58.212770 26727 tester_helper.h:186] Run 1 times...
mkldnn_verbose,exec,convolution,jit:avx2,forward_inference,fsrc:nchw fwei:Ohwi8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic3oc64_ih207oh101kh7sh2dh0ph0_iw175ow85kw7sw2dw0pw0,2.58203
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,0.0581055
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x64x101x85,2.18994
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x64x101x85,0.559082
mkldnn_verbose,exec,convolution,jit:avx2,forward_inference,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic64oc128_ih101oh49kh5sh2dh0ph0_iw85ow41kw5sw2dw0pw0,12.998
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.0461426
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x128x49x41,0.291016
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x128x49x41,0.280029
mkldnn_verbose,exec,convolution,jit:avx2,forward_inference,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic128oc256_ih49oh23kh5sh2dh0ph0_iw41ow19kw5sw2dw0pw0,13.1311
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.130859
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x256x23x19,0.101807
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,1x256x23x19,0.098877
mkldnn_verbose,exec,convolution,jit:avx2,forward_inference,fsrc:nChw8c fwei:OIhw8i8o fbia:x fdst:nChw8c,alg:convolution_direct,mb1_g1ic256oc512_ih23oh11kh3sh2dh0ph0_iw19ow9kw3sw2dw0pw0,4.81299
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.0859375
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,1x512x11x9,0.0471191
mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:blocked fbia:undef fdst:nchw,alg:deconvolution_direct,mb1_g1ic512oc512_ih11oh23kh3sh2dh0ph0_iw9ow19kw3sw2dw0pw0,14.6499
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic512ih23iw19,1.35791
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic512ih23iw19,0.0678711
mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:blocked fbia:undef fdst:nchw,alg:deconvolution_direct,mb1_g1ic512oc256_ih23oh49kh5sh2dh0ph0_iw19ow41kw5sw2dw0pw0,60.1091
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic256ih49iw41,3.09106
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic256ih49iw41,0.151123
mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:blocked fbia:undef fdst:nchw,alg:deconvolution_direct,mb1_g1ic256oc128_ih49oh101kh5sh2dh0ph0_iw41ow85kw5sw2dw0pw0,70.571
mkldnn_verbose,exec,batch_normalization,ncsp_bnorm:any,forward_inference,fdata:nchw fdiff:undef,flags:3,mb1ic128ih101iw85,6.60815
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_relu,mb1ic128ih101iw85,0.390869
mkldnn_verbose,exec,deconvolution,ref:any,forward_inference,fsrc:nchw fwei:blocked fbia:undef fdst:nchw,alg:deconvolution_direct,mb1_g1ic128oc3_ih101oh207kh7sh2dh0ph0_iw85ow175kw7sw2dw0pw0,8.59497
mkldnn_verbose,exec,eltwise,jit:avx2,forward_inference,fdata:nchw fdiff:undef,alg:eltwise_tanh,mb1ic3ih207iw175,0.156006
I1128 21:34:58.423396 26727 helper.h:229] ====== batch_size: 1, repeat: 1, threads: 1, thread id: 0, latency: 210.595ms, fps: 4.74845 ======
[       OK ] Analyzer_resnet50.profile_mkldnn (639 ms)
```

**Comment by luotao1** [comment]
@jianhang-liu @jczaja If all the CI pass, could this PR be merged at first? If you have further improvment, you can create a new PR.

**Comment by jczaja** [comment]
@luotao1 Yes, please merge it. We will look further , based on log you provided if we can improve.

**Review by luotao1 — APPROVED** [review]
LGTM! Thanks very much!

**Pull request merged**